system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -15,27 +15,39 @@ const TopSiteLink = props => {
   let showSmallFavicon = false;
   let smallFaviconStyle;
   let smallFaviconFallback;
-  if (tippyTopIcon || faviconSize >= MIN_RICH_FAVICON_SIZE) {
-    // styles and class names for top sites with rich icons
-    imageClassName = "top-site-icon rich-icon";
-    imageStyle = {
-      backgroundColor: link.backgroundColor,
-      backgroundImage: `url(${tippyTopIcon || link.favicon})`
-    };
-  } else {
-    // styles and class names for top sites with screenshot + small icon in top left corner
-    imageClassName = `screenshot${link.screenshot ? " active" : ""}`;
-    imageStyle = {backgroundImage: link.screenshot ? `url(${link.screenshot})` : "none"};
+  // If the screenshot request failed use letter fallback.
+  if (!screenshotRequestFailed) {
```

Comment:
Something about this seems unnecessarily complex,but I'll need to think about it a bit more. I suppose we should be able to avoid all the lines of change by having a empty
```js
if (screenshotRequestFailed) {
  // Just use the letter fallback
} else if (screenshotPreview…) {
} else if (tippyTopIcon …) {
```

Reply:
Changed as suggested.